### ISSUE TITLE
sklearn-xarray compatability

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask-ml#112]
There was a question over in https://github.com/pangeo-data/pangeo/issues/61#issuecomment-356113485 about how dask-ml works with [sklearn-xarray](https://phausamann.github.io/sklearn-xarray/)

When I first tried it out, things like `dask_ml.preprocessing.StandardScalar()` failed since we do `X.mean(0)`, instead of `X.mean(axis=0)`. I think I tried fixing that locally and a small example worked, though my memory is hazy.
 
I don't personally have much experience with xarray , but would certainly welcome PRs to make things work smoothly.

cc @phausamann , @gmaze

**Comment by phausamann** [comment]
I've looked into it and I believe that for the simple case of using `dask-ml` estimators on the underlying data of `DataArrays` and `Datasets` it is sufficient to make sure the wrapper always passes the `.data` attribute to the estimator.

I'll push a commit to the `develop` branch of `sklearn-xarray` later today that should make the following code work:

    from sklearn_xarray import wrap
    from dask_ml.preprocessing import StandardScaler
    import xarray as xr
    import numpy as np
    import dask.array as da
    
    X = xr.DataArray(
            da.from_array(np.random.random((100, 10)), chunks=(10, 10)),
            coords={'sample': range(100), 'feature': range(10)},
            dims=('sample', 'feature')
        )
    
    Xt = wrap(StandardScaler()).fit_transform(X)

    In []: type(Xt.data)
    Out[]: dask.array.core.Array

Now, for compatibility with fully dasked-backed `Datasets` there's a lot more to be done in order to make sure that on-disk coordinates are not completely loaded into memory during the coordinate update...

**Comment by gmaze** [comment]
Having dask-ml and sklearn-xarray "compatible" would be a great plus for both libraries as it would allow a simple mechanism for scaling stat methods to large datasets !

**Comment by phausamann** [comment]
In the end, I even made a new release (0.1.3) which is available through pip as of now.

With this update, the above code should work, as well as (hopefully) other dask-ml estimators. Feel free to open an issue in the sklearn-xarray repository if anything doesn't work as expected.

**Comment by TomAugspurger** [comment]
Thanks @phausamann, I tried with

```python
import xarray as xr
from sklearn_xarray import wrap
from sklearn_xarray.data import load_dummy_dataarray
from dask_ml.preprocessing import StandardScaler

X = load_dummy_dataarray()
X.to_netcdf("/tmp/X.nc")
X2 = xr.open_mfdataset("/tmp/X.nc").to_array()[0]

X3 = xr.concat([X2] * 1000, dim='sample')

Xt = wrap(StandardScaler()).fit_transform(X3)
Xt
```

And everything seemed to work correctly.
